src/middlewared/middlewared/plugins/network.py

```diff
@@ -480,7 +480,10 @@ class DNSService(Service):
         else:
             gc = await self.middleware.call('datastore.query', 'network.globalconfiguration', None, {'get': True})
             if gc['gc_domain']:
-                domain = gc['gc_domain']
+                domains.append('{}\n'.format(gc['gc_domain']))
```

Comment:
I think you misunderstood me here.

Each item of the domain list, must be a domain, no spaces or new lines.